src/Components/Web.JS/src/Boot.Server.ts

```diff
@@ -163,3 +164,18 @@ Blazor.start = boot;
 if (shouldAutoStart()) {
   boot();
 }
+
+function showUnavailableTransportErrorMessage(ex: Error) {
+  const unableToConnectTransportMessage = "Unable to connect to the server with any of the available transports.";
+  if (!ex.message.startsWith(unableToConnectTransportMessage)) {
+    return;
+  }
+
+  if (ex.message.includes(`'WebSockets' is not supported in your environment.`)) {
+    showErrorNotification('Unable to connect, please ensure you are using an updated browser and WebSockets are available.');
+  } if (ex.message.includes(`'${HttpTransportType[HttpTransportType.WebSockets]}' is disabled by the client.`)) {
+    showErrorNotification('Unable to connect, please enable WebSockets on the client.');
+  } else if (ex.message.includes(`'${HttpTransportType[HttpTransportType.LongPolling]}' is disabled by the client.`)) {
+    showErrorNotification('Unable to connect, please enable LongPolling on the client.');
```

Comment:
These last two error messages aren't easily recoverable from by the end-user. In which case, we should just display the generic `Unhandled exception occurred` message (thoughts?). Developers will still be able to see what went wrong in the developer console.

Reply:
Removed these two. https://github.com/dotnet/aspnetcore/commit/e03d084445821d142d144a54c535658716fc0f20#diff-7c47c684c78e3f5b5eeecb28b249b03afcefcd90a925cc20b69ec2cdec261752L176-L180